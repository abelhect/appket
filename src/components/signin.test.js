import React from 'react';
import {shallow} from 'enzyme';
import SignIn from './signin';

describe('The SignIn Component', ()=>{
  const signin = shallow(<SignIn />);
  it('renders correctly', ()=>{
    expect(signin).toMatchSnapshot();
  });

  it('displays the title', ()=>{
    expect(signin.find('.title').text()).toEqual('Sign In');
  });

  it('creates an input for the user email', () => {
    expect(signin.find('.form-control').exists()).toBe(true);
  });

  it('creates an input for the user password', ()=>{
    expect(signin.find('.form-control').exists()).toBe(true);
  });

  describe('performs some things when user click `Sign In` button', ()=>{
    // beforeEach(()=> signin.find('.btn-signin').simulate('click'));
    it('creates a button for user to sign in', ()=>{
      expect(signin.find('.btn').simulate('click'));

    });
  });

// test for error state update
  // it('displays an error message when failed sign in', ()=>{
  //   expect(signin.find('.error-msg').text()).toEqual('Error Msg Placeholder!');
  // });

  // it('creates a link to `Sign Up` if user is not a member yet', ()=>{
  //   expect(signin.find('.signup-link').text()).toEqual('Not a member yet? Sign Up Here!');
  // });
});
