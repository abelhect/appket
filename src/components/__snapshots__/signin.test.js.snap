// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SignIn Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
>
          Sign In
</h2>,
        <div
          className="form-group"
>
          <input
                    className="form-control"
                    onChange={[Function]}
                    placeholder="email"
                    style={
                              Object {
                                        "marginRight": "5px",
                                      }
                    }
                    type="email"
          />
          <input
                    className="form-control"
                    onChange={[Function]}
                    placeholder="password"
                    style={
                              Object {
                                        "marginRight": "5px",
                                      }
                    }
                    type="password"
          />
          <button
                    className="btn btn-primary"
                    onClick={[Function]}
          >
                    Sign In
          </button>
</div>,
        <div
          className="error-msg"
          style={
                    Object {
                              "margin": "1%",
                            }
          }
/>,
        <div
          className="signup-link"
>
          Not a member yet? Sign Up Here!
</div>,
      ],
      "className": "form-inline",
      "style": Object {
        "margin": "5%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
          "className": "title",
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="email"
              style={
                            Object {
                                          "marginRight": "5px",
                                        }
              }
              type="email"
/>,
            <input
              className="form-control"
              onChange={[Function]}
              placeholder="password"
              style={
                            Object {
                                          "marginRight": "5px",
                                        }
              }
              type="password"
/>,
            <button
              className="btn btn-primary"
              onClick={[Function]}
>
              Sign In
</button>,
          ],
          "className": "form-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "onChange": [Function],
              "placeholder": "email",
              "style": Object {
                "marginRight": "5px",
              },
              "type": "email",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "onChange": [Function],
              "placeholder": "password",
              "style": Object {
                "marginRight": "5px",
              },
              "type": "password",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign In",
              "className": "btn btn-primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Sign In",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "error-msg",
          "style": Object {
            "margin": "1%",
          },
        },
        "ref": null,
        "rendered": "",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Not a member yet? Sign Up Here!",
          "className": "signup-link",
        },
        "ref": null,
        "rendered": "Not a member yet? Sign Up Here!",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
>
            Sign In
</h2>,
          <div
            className="form-group"
>
            <input
                        className="form-control"
                        onChange={[Function]}
                        placeholder="email"
                        style={
                                    Object {
                                                "marginRight": "5px",
                                              }
                        }
                        type="email"
            />
            <input
                        className="form-control"
                        onChange={[Function]}
                        placeholder="password"
                        style={
                                    Object {
                                                "marginRight": "5px",
                                              }
                        }
                        type="password"
            />
            <button
                        className="btn btn-primary"
                        onClick={[Function]}
            >
                        Sign In
            </button>
</div>,
          <div
            className="error-msg"
            style={
                        Object {
                                    "margin": "1%",
                                  }
            }
/>,
          <div
            className="signup-link"
>
            Not a member yet? Sign Up Here!
</div>,
        ],
        "className": "form-inline",
        "style": Object {
          "margin": "5%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "className": "title",
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="email"
                style={
                                Object {
                                                "marginRight": "5px",
                                              }
                }
                type="email"
/>,
              <input
                className="form-control"
                onChange={[Function]}
                placeholder="password"
                style={
                                Object {
                                                "marginRight": "5px",
                                              }
                }
                type="password"
/>,
              <button
                className="btn btn-primary"
                onClick={[Function]}
>
                Sign In
</button>,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "onChange": [Function],
                "placeholder": "email",
                "style": Object {
                  "marginRight": "5px",
                },
                "type": "email",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "onChange": [Function],
                "placeholder": "password",
                "style": Object {
                  "marginRight": "5px",
                },
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign In",
                "className": "btn btn-primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Sign In",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "error-msg",
            "style": Object {
              "margin": "1%",
            },
          },
          "ref": null,
          "rendered": "",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Not a member yet? Sign Up Here!",
            "className": "signup-link",
          },
          "ref": null,
          "rendered": "Not a member yet? Sign Up Here!",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
